### PR TITLE
Expanding tasks model

### PR DESCRIPTION
**Pull request by olsona**
Expanded tasks model to include completion time and due time.
Added element to TaskSeries POJO to represent RRules, as found in recurring events in RTM.

### CODE REVIEW

**Comment by olsona** [comment]
Starts to address #346, #432.

File: extensions/data-transfer/portability-data-transfer-google/src/main/java/org/dataportabilityproject/datatransfer/google/tasks/GoogleTasksExporter.java [hunk 1]
```diff
@@ -106,7 +107,9 @@ public GoogleTasksExporter(GoogleCredentialFactory credentialFactory) {
         result
             .getItems()
             .stream()
-            .map(t -> new TaskModel(resource.getId(), t.getTitle(), t.getNotes()))
+            .map(t -> new TaskModel(resource.getId(), t.getTitle(), t.getNotes(),
```
Comment (holachuy) [inline extensions/data-transfer/portability-data-transfer-google/src/main/java/org/dataportabilityproject/datatransfer/google/tasks/GoogleTasksExporter.java]: Do we need to validate these values exist?

File: extensions/data-transfer/portability-data-transfer-google/src/main/java/org/dataportabilityproject/datatransfer/google/tasks/GoogleTasksExporter.java [hunk 2]
```diff
@@ -106,7 +107,9 @@ public GoogleTasksExporter(GoogleCredentialFactory credentialFactory) {
...
+                Instant.ofEpochMilli(t.getCompleted().getValue()),
```
Comment (holachuy) [inline extensions/data-transfer/portability-data-transfer-google/src/main/java/org/dataportabilityproject/datatransfer/google/tasks/GoogleTasksExporter.java]: What if t.getCompleted() or getDue() are null? Should we write some defensive code when fetching values?

File: extensions/data-transfer/portability-data-transfer-rememberthemilk/src/main/java/org/dataportabilityproject/transfer/rememberthemilk/tasks/RememberTheMilkService.java [hunk 3]
```diff
@@ -17,6 +17,7 @@
 
 import static com.google.common.base.Preconditions.checkState;
 
+import com.fasterxml.jackson.databind.DeserializationFeature;
```
Comment (holachuy) [inline extensions/data-transfer/portability-data-transfer-rememberthemilk/src/main/java/org/dataportabilityproject/transfer/rememberthemilk/tasks/RememberTheMilkService.java]: Is this used?

File: extensions/data-transfer/portability-data-transfer-rememberthemilk/src/main/java/org/dataportabilityproject/transfer/rememberthemilk/tasks/RememberTheMilkTasksExporter.java [hunk 4]
```diff
@@ -40,13 +43,16 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /*
  * Exporter for Tasks data type from Remember The Milk Service.
  */
 public class RememberTheMilkTasksExporter implements Exporter<AuthData, TaskContainerResource> {
 
   private final AppCredentials appCredentials;
+  Logger logger = LoggerFactory.getLogger(RememberTheMilkTasksExporter.class);
```
Comment (holachuy) [inline extensions/data-transfer/portability-data-transfer-rememberthemilk/src/main/java/org/dataportabilityproject/transfer/rememberthemilk/tasks/RememberTheMilkTasksExporter.java]: private static final

File: extensions/data-transfer/portability-data-transfer-rememberthemilk/src/main/java/org/dataportabilityproject/transfer/rememberthemilk/tasks/RememberTheMilkTasksExporter.java [hunk 5]
```diff
@@ -95,9 +101,17 @@ private ExportResult exportTask(
         for (TaskSeries taskSeries : taskList.taskseries) {
           // TODO: figure out what to do with notes
           String notesStr = taskSeries.notes == null ? "" : taskSeries.notes.toString();
-          tasks.add(new TaskModel(oldListId, taskSeries.name, notesStr));
           for (Task task : taskSeries.tasks) {
-            // TODO: handle completion date, but its odd there can be more than one.
+            // TODO: What do we actually want to do with this?
```
Comment (holachuy) [inline extensions/data-transfer/portability-data-transfer-rememberthemilk/src/main/java/org/dataportabilityproject/transfer/rememberthemilk/tasks/RememberTheMilkTasksExporter.java]: Does this comment apply or does it have to be more specific?

File: extensions/data-transfer/portability-data-transfer-rememberthemilk/src/main/java/org/dataportabilityproject/transfer/rememberthemilk/tasks/RememberTheMilkTasksExporter.java [hunk 6]
```diff
@@ -95,9 +101,17 @@ private ExportResult exportTask(
...
+            Instant completedTime = null;
+            Instant dueTime = null;
+            if (!Strings.isNullOrEmpty(task.completed)) {
+              completedTime = Instant.parse(task.completed);
```
Comment (holachuy) [inline extensions/data-transfer/portability-data-transfer-rememberthemilk/src/main/java/org/dataportabilityproject/transfer/rememberthemilk/tasks/RememberTheMilkTasksExporter.java]: What if completed or due are nuill?

File: extensions/data-transfer/portability-data-transfer-google/src/main/java/org/dataportabilityproject/datatransfer/google/tasks/GoogleTasksExporter.java (hunk 2 above)
Comment (olsona) [reply extensions/data-transfer/portability-data-transfer-google/src/main/java/org/dataportabilityproject/datatransfer/google/tasks/GoogleTasksExporter.java]: Added ternary operators.

File: extensions/data-transfer/portability-data-transfer-rememberthemilk/src/main/java/org/dataportabilityproject/transfer/rememberthemilk/tasks/RememberTheMilkService.java (hunk 3 above)
Comment (olsona) [reply extensions/data-transfer/portability-data-transfer-rememberthemilk/src/main/java/org/dataportabilityproject/transfer/rememberthemilk/tasks/RememberTheMilkService.java]: Done.

File: extensions/data-transfer/portability-data-transfer-rememberthemilk/src/main/java/org/dataportabilityproject/transfer/rememberthemilk/tasks/RememberTheMilkTasksExporter.java (hunk 4 above)
Comment (olsona) [reply extensions/data-transfer/portability-data-transfer-rememberthemilk/src/main/java/org/dataportabilityproject/transfer/rememberthemilk/tasks/RememberTheMilkTasksExporter.java]: Done.

File: extensions/data-transfer/portability-data-transfer-rememberthemilk/src/main/java/org/dataportabilityproject/transfer/rememberthemilk/tasks/RememberTheMilkTasksExporter.java (hunk 6 above)
Comment (olsona) [reply extensions/data-transfer/portability-data-transfer-rememberthemilk/src/main/java/org/dataportabilityproject/transfer/rememberthemilk/tasks/RememberTheMilkTasksExporter.java]: Done.

File: extensions/data-transfer/portability-data-transfer-rememberthemilk/src/main/java/org/dataportabilityproject/transfer/rememberthemilk/tasks/RememberTheMilkTasksExporter.java (hunk 5 above)
Comment (olsona) [reply extensions/data-transfer/portability-data-transfer-rememberthemilk/src/main/java/org/dataportabilityproject/transfer/rememberthemilk/tasks/RememberTheMilkTasksExporter.java]: Clarified.

File: extensions/data-transfer/portability-data-transfer-google/src/main/java/org/dataportabilityproject/datatransfer/google/tasks/GoogleTasksExporter.java (hunk 1 above)
Comment (olsona) [reply extensions/data-transfer/portability-data-transfer-google/src/main/java/org/dataportabilityproject/datatransfer/google/tasks/GoogleTasksExporter.java]: Worked fine with empty-titled and empty-note tasks, so I think we don't.

**Pull request merged**